ksqldb-engine/src/test/java/io/confluent/ksql/structured/SchemaKSourceFactoryTest.java

```diff
@@ -239,12 +248,243 @@ public void shouldBuildCorrectFormatsForV2NonWindowedTable() {
+        dataSource,
+        contextStacker
+    );
+
+    // Then:
+    assertThat(((StreamSource) result.getSourceStep()).getPseudoColumnVersion(), equalTo(NEW_PSEUDOCOLUMN_VERSION));
+    assertValidSchema(result);
+  }
+
+  @Test
+  public void shouldReplaceOldWindowedStreamSourceWithMatchingPseudoColumnVersion() {
+    // Given:
+    givenWindowedStream();
+    givenExistingQueryWithOldPseudoColumnVersion();
+    givenOnNewPseudoColumnVersion();
+
+    // When:
+    final SchemaKStream<?> result = SchemaKSourceFactory.buildSource(
+        buildContext,
+        dataSource,
+        contextStacker
+    );
+
+    // Then:
+    assertThat(((WindowedStreamSource) result.getSourceStep()).getPseudoColumnVersion(), equalTo(OLD_PSEUDOCOLUMN_VERSION));
+    assertValidSchema(result);
+  }
+
+  @Test
+  public void shouldCreateWindowedStreamSourceWithNewPseudoColumnVersionIfNoOldQuery() {
+    // Given:
+    givenWindowedStream();
+    givenOnNewPseudoColumnVersion();
+
+    // When:
+    final SchemaKStream<?> result = SchemaKSourceFactory.buildSource(
+        buildContext,
+        dataSource,
+        contextStacker
+    );
+
+    // Then:
+    assertThat(((WindowedStreamSource) result.getSourceStep()).getPseudoColumnVersion(), equalTo(NEW_PSEUDOCOLUMN_VERSION));
+    assertValidSchema(result);
+  }
+
+  @Test
+  public void shouldCreateNonWindowedTableSourceV2WithNewPseudoColumnVersionIfNoOldQuery() {
+    // Given:
+    givenNonWindowedTable();
+    givenOnNewPseudoColumnVersion();
+
+    // When:
+    final SchemaKStream<?> result = SchemaKSourceFactory.buildSource(
+        buildContext,
+        dataSource,
+        contextStacker
+    );
+
+    // Then:
+    assertThat(((TableSource) result.getSourceStep()).getPseudoColumnVersion(), equalTo(NEW_PSEUDOCOLUMN_VERSION));
+    assertValidSchema(result);
+  }
+
+  @Test
+  public void shouldCreateNonWindowedTableSourceV1WithOldPseudoColumnVersionIfNoOldQuery() {
+    // Given:
+    givenNonWindowedTable();
+    givenOnOldPseudoColumnVersion();
+
+    // When:
+    final SchemaKStream<?> result = SchemaKSourceFactory.buildSource(
+        buildContext,
+        dataSource,
+        contextStacker
+    );
+
+    // Then:
+    assertThat(((TableSourceV1) result.getSourceStep()).getPseudoColumnVersion(), equalTo(OLD_PSEUDOCOLUMN_VERSION));
+    assertValidSchema(result);
+  }
+
+  @Test
+  public void shouldReplaceTableSourceV1WithSame() {
+    // Given:
+    givenNonWindowedTable();
+    givenExistingTableSourceV1();
+    givenOnNewPseudoColumnVersion();
+
+    // When:
+    final SchemaKStream<?> result = SchemaKSourceFactory.buildSource(
+        buildContext,
+        dataSource,
+        contextStacker
+    );
+
+    // Then:
+    assertThat(((TableSourceV1) result.getSourceStep()).getPseudoColumnVersion(), equalTo(OLD_PSEUDOCOLUMN_VERSION));
+    assertValidSchema(result);
+  }
+
+  @Test
+  public void shouldReplaceOldWindowedTableSourceWithMatchingPseudoColumnVersion() {
```

Comment:
```suggestion
  public void shouldReplaceWindowedTableSourceWithMatchingPseudoColumnVersion() {
```

Reply:
Why this but not the others with the same convention (ie `shouldReplaceOldWindowedStreamSourceWithMatchingPseudoColumnVersion`)? I don't feel strongly that the `Old` is necessary in any, though.

Reply:
I'm all for updating the others too! 😁

Reply:
I think "old" is unnecessary/confusing because:

* there's no such thing as an "old" windowed table source, i.e., there's only WindowedTableSourceV1 and no WindowedTableSourceV2
* if "old" is meant to mean "existing" rather than "of an old version", then it is redundant because if it's being replaced then it must be old
* table sources actually do have two versions (TableSourceV1 and TableSourceV2) so "old" does make sense in this context (though ideally we'd say TableSourceV1 and TableSourceV2 rather than OldTableSource and NewTableSource, in order to age better). Using "old" in places where it doesn't make sense detracts from places where it does make sense.